### PR TITLE
153 add reddit service

### LINKED ISSUE #153: Add Reddit Service
**Short description:**
Add the service Reddit, including OAuth connection, front and back features

### PR DESCRIPTION
**Pull request by tbellicha**
# Purpose

The purpose of this pull request is to add the service Reddit.

Fixes #153

## Added features

* This service has been added to the application. It includes:
  * A reaction which subscribe to a subreddit.
  * A reaction which subscribe to the newest subreddit.
  * A reaction which unsubscribe to a subreddit.

## Fixed feature

* No limitation on Deezer reaction fixed.

### CODE REVIEW

File: server/server_app.js [hunk 1]
```diff
@@ -1106,6 +1150,50 @@ require('./api/area/area.js')(app, passport, database)
 require('./api/area/reaction/reaction.js')(app, passport, database)
 require('./api/area/action/action.js')(app, passport, database)
 
+/**
+ * Generate token thanks to a code in Reddit
```
Comment (LucasTesnier) [inline server/server_app.js:2711]: missing doc

File: server/services/reddit/init.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ */
+const createRedditService = async () => {
+  try {
+    const Reddit = await database.prisma.Service.create({
+      data: {
+        name: 'Reddit',
+        description: 'Reddit service',
+        primaryColor: '#FF4500',
+        secondaryColor: '#FFFFFF',
+        icon: '',
```
Comment (LucasTesnier) [inline server/services/reddit/init.js]: Missing icon

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Missing doc and add reddit icon otherwise NTR goat

File: application/app_code/lib/pages/auth_linker/auth_linker_functional.dart [hunk 3]
```diff
@@ -208,13 +218,84 @@ Future<String> getDiscordToken() async {
...
+    String? tokenSave = '${redditAuthBox.token}';
+    redditAuthBox.token = null;
+    String? error = await publishNewToken();
+    if (error != null) {
+      redditAuthBox.token = tokenSave;
+      return error;
+    }
+    redditAuthBox.isEnable = false;
+  } else {
+    String clientId = "7P3NMqftCBgr7H-XaPUbNg";
```
Comment (ZQUEMA) [inline application/app_code/lib/pages/auth_linker/auth_linker_functional.dart:233]: What's this hard ID ?

File: server/server_app.js [hunk 4]
```diff
@@ -163,6 +164,43 @@ app.post('/api/signup', (req, res, next) => {
...
+ * @param {*} userId The ID of the user in the database
+ * @param {*} refresh_token The refresh token of the user
+ */
+async function refresh_reddit_token (userId, refresh_token) {
+  try {
+    const postData = {
+      grant_type: 'refresh_token',
+      refresh_token: refresh_token
+    }
+    const authString = `7P3NMqftCBgr7H-XaPUbNg:gPFD-f_P2DNz8WVjm-Qwj21G8hS9KA`
```
Comment (ZQUEMA) [inline server/server_app.js:287]: What's is this hard auth token ?

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
You have to answer the questions

**Review by TomDUVAL-MAHE — DISMISSED** [review]

**Review by ZQUEMA — DISMISSED** [review]

**Review by LucasTesnier — DISMISSED** [review]

**Review by TomDUVAL-MAHE — DISMISSED** [review]

**Review by LucasTesnier — APPROVED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Pull request merged**